orc8r/cloud/go/tools/swaggergen/generate/generate.go

```diff
@@ -242,6 +288,17 @@ func StripAndWriteSwaggerSpecs(specs map[string]MagmaSwaggerSpec) error {
 	return nil
 }
 
+// RetrieveServiceName parses the targetFilepath for its originating service.
+func RetrieveServiceName(targetFilePath string, rootDir string) (string, error) {
+	absTargetFilepath, err := filepath.Abs(targetFilePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "target filepath %s is invalid", targetFilePath)
+	}
+	splitPath := strings.Split(absTargetFilepath, "/")
+	outFile := splitPath[len(splitPath)-4] + ".swagger.v1.yml"
```

Comment:
This is too magic. Add a comment describing what/why you're grabbing that part of the filepath